.github/workflows/main.yml

```diff
@@ -23,15 +23,15 @@ jobs:
 
     steps:
 
-     -  name: Check out source code
```

Comment:
Please squash these changes with the first patch that adds `main.yml`.

Reply:
This change contains code from commits after the first patch.
Merge conflicts are there when trying to squash.

Reply:
OK, I mean the indentation fixes could be done in the `github: Create main.yml` commit to make the last commit do nothing, and then it'll disappear.

If it's causing too much git conflicts then no worries, we can merge it like this.

Reply:
Yes, this commit is on top of the two commits:
https://github.com/kernelci/kernelci-pipeline/pull/4/commits/82af750d1cfd794fea7714ab41322f6bdf9169f1
https://github.com/kernelci/kernelci-pipeline/pull/4/commits/49d7a9053878cf993afa985fa38d6ed16b8ecb7f

That's why it's difficult to squash it.
I can force push those two commits with code changes and delete this commit, or we can merge it like this.

Reply:
Yeah that's what I meant, it should be possible to move the changes from the last commit to the earlier ones.  I had a look locally, here's how I did it:
```
$ git checkout jeny/setup-github-actions -b tmp
$ cp .github/workflows/main.yml _main.yml
$ git rebase -i main --onto=main
# edit all the commits that change the .yml file
# then for each commit:
$ diff -u .github/workflows/main.yml _main.yml
# to see the differences and adjust the file accordingly, you can even copy things from _main.yml. Then:
$ git add config/core/build-configs-staging.yaml
$ git rebase --continue
# eventually:
nothing added to commit but untracked files present
Could not apply 8d1fcdb2315e... github/main.yaml: Fix indentation
$ git rebase --continue
```
And then `git diff jeny/setup-github-actions` shows no difference, so the resulting code is the same but without the last commit to fix the indent.  I've pushed my resulting branch here, feel free to pick commits from it and force-push to this PR or do the same things by hand:
https://github.com/gctucker/kernelci-pipeline/commits/jeny-setup-github-actions

I believe the indentation matters in this YAML file, I haven't tried to parse it but it's better if we can avoid having any commits with issues if we can.

Reply:
Updated commits.

Reply:
Thank you for the guide @gctucker :)